### PR TITLE
Completed sgt-back-end exercise.

### PR DESCRIPTION
**Pull request by arjunkahlon**
Aug-25-2022 17-52-57

### CODE REVIEW

**Comment by arjunkahlon** [comment]
Aug-25-2022 17-57-57

File: sgt-back-end/index.js [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    return res.status(200).json(grades);
+  }).catch(err => {
+    console.error(err);
+    return res.status(500).json({ error: 'An unexpected error occured' });
+  });
+});
+
+app.post('/api/grades', (req, res, next) => {
+  if (!req.body.name || !req.body.course || !req.body.score) {
+    return res.status(400).json({ Error: 'Missing name, course, or score.' });
```
Comment (TimelikeClosure) [inline sgt-back-end/index.js:67]: Zero is a valid score

File: sgt-back-end/index.js [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+  const sql = `delete from "grades"
+                  where "gradeId" = $1
+                returning *
+              `;
+
+  const params = [Number(gradeId)];
+  db.query(sql, params).then(result => {
+    const deletedGrade = result.rows[0];
+    if (!deletedGrade) {
+      return res.status(400).json({ Error: 'Id does not exist in database' });
```
Comment (TimelikeClosure) [inline sgt-back-end/index.js:134]: This is the wrong status code for a non-existent id

**Review by TimelikeClosure — CHANGES_REQUESTED** [review]

File: sgt-back-end/index.js (hunk 1 above)
Comment (TimelikeClosure) [reply sgt-back-end/index.js:67]: To be clear, a `score` of `0` _should_ be treated as valid. Right now, it's triggering a `400 Bad Request`.

**Review by TimelikeClosure — CHANGES_REQUESTED** [review]

File: sgt-back-end/index.js [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+app.put('/api/grades/:gradeId', (req, res, next) => {
+  const gradeId = Number(req.params.gradeId);
+  if (!Number.isInteger(gradeId) || gradeId <= 0) {
+    return res.status(400).json({
+      error: '"gradeId" must be a positive integer'
+    });
+  }
+  if (!req.body.name || !req.body.course || !req.body.score) {
+    return res.status(400).json({ Error: 'Missing name, course, or score.' });
```
Comment (TimelikeClosure) [inline sgt-back-end/index.js:94]: A `score` of `0` should not result in a `400 Bad Request`.

**Review by TimelikeClosure — CHANGES_REQUESTED** [review]

**Review by TimelikeClosure — APPROVED** [review]

**Pull request merged**